srml/sudo/src/lib.rs

```diff
@@ -14,8 +14,112 @@
 // You should have received a copy of the GNU General Public License
 // along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
 
-//! The Example: A simple example of a runtime module demonstrating
-//! concepts, APIs and structures common to most runtime modules.
+//! # Sudo Module
+//!
+//! ## Overview
+//! 
+//! The Sudo module allows for a single account (called the "sudo key")
+//! to execute dispatchable functions that require a `Root` call
+//! or designate a new account to replace them as the sudo key.
+//! Only one account can be the sudo key at a time.
+//! 
+//! You can start using the Sudo module by implementing the sudo [`Trait`].
+//! 
+//! Supported dispatchables functions are documented as part of the [`Call`] enum.
+//! 
+//! ## Interface
+//! 
+//! ### Dispatchable
+//! 
+//! Only the sudo key can call the dispatchable functions from the Sudo module.
+//! 
+//! * `sudo` - Make a `Root` call to a dispatchable function.
+//! * `set_key` - Assign a new account to be the key of the superuser.
+//! 
+//! Please refer to the [`Call`] enum and its associated variants for a detailed list of dispatchable functions.
+//!
+//! ## Usage
+//! 
+//! ### Prerequisites
+//! 
+//! To use the Sudo module in your runtime, you must implement the following trait in your runtime:
+//! 
+//! ```ignore
+//! impl sudo::Trait for Runtime {
+//! 	/// The ubiquitous event type.
+//! 	type Event = Event;
+//! 	type Proposal = Call;
+//! }
+//! ```
+//! 
+//! You can then import the Sudo module in your `construct_runtime!` macro with:
+//! 
+//! ```ignore
+//! Sudo: sudo,
+//! ```
+//! 
+//! ### Executing Privileged Functions
+//! 
+//! The Sudo module itself is not intended to be used within other modules.
+//! Instead, you can build "privileged functions" in other modules that require `Root` origin.
+//! You can execute these privileged functions by calling `sudo()` with the sudo key account.
+//! Privileged functions cannot be directly executed via an extrinsic.
+//! 
+//! Learn more about privileged functions and `Root` origin in the [`Origin`] type documentation.
+//! 
+//! ### Simple Code Snippet
+//! 
+//! This is an example of a module that exposes a privileged function:
+//! 
+//! ```ignore
+//! use support::{decl_module, dispatch::Result};
+//! use system::ensure_root;
+//! 
+//! pub trait Trait: system::Trait {}
+//! 
+//! decl_module! {
+//!     pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+//!         pub fn privileged_function(origin) -> Result {
+//!             ensure_root(origin)?;
+//!             
+//!             // do something...
+//! 
+//!             Ok(())
+//!         }
+//!     }
+//! }
+//! ```
+//! 
+//! ### Example from SRML
+//! 
+//! The consensus module exposes a `set_code()` privileged function
+//! that allows you to set the on-chain Wasm runtime code:
+//! 
+//! ```ignore
+//! /// Set the new code.
+//! pub fn set_code(new: Vec<u8>) {
```

Comment:
why doesn't it need to check `ensure_root(origin)?;` I just documented `consensus` and am seeking the answer to this question.